client/src/context/quiz-context.type.ts

```diff
@@ -0,0 +1,34 @@
+import { Dispatch } from "react";
+import { Option, Quiz } from "../database/data.type";
+
+export type VALUE_TYPE = {
+  question: Quiz;
+
+  selectedOption: Option | null;
+};
+export type ACTION_TYPE =
+  | {
+      type: "NEXT_QUESTION";
+      value: VALUE_TYPE;
+    }
+  | { type: "WRONG_ANSWER" }
+  | { type: "RESET" }
+  | { type: "SET_CATEGORY"; value: { category: string } }
+  | { type: "SET_USERNAME"; value: { userName: string } }
+  | { type: "SKIP_QUESTION" }
+  | { type: "SET_QUIZ"; value: { quiz: Quiz[]; category: string[] } }
+  | { type: "SET_CURRENT_QUIZ"; value: { quiz: Quiz[]; category: string } };
```

Comment:
- General convention for `action` object is to have `type` and `payload` as properties.
- Sometimes using `value` can be confusing, but this is just a suggestion I don't have any strong opinion on this, it's upto you.